frontend/test/account-send.test.ts

```diff
@@ -0,0 +1,561 @@
+      expect(decryptedData).toEqual({
+        advancedMode,
+        usePublicKeyChecked,
+        address: recipientAddress,
+        pubKey: `0x99${pubKey.slice(4, 4 + 22)}`,
+      });
+    }
+  });
+
+  it('Encrypts and decrypts properly for values with leading zeros', () => {
+    // `BigNumber.from(myHexString).toHexString()` where `myHexString` has leading zero bytes will
+    // truncate those leading zeroes from the string. If data is not manually zero-padded to the
+    // correct length, this can cause a bug where the leading zero bytes of the address are
+    // truncated and trailing bytes of other data are appended. The above fuzz test found this bug,
+    // and this is a concrete test with known inputs would fail if the bug were present.
+    const recipientAddress = '0x00bB833415cf56f112389E146Cb7847dDbE93fB5';
+    const pubKey =
+      '0x040426c84897f9e07632687f69ef714090307dcdc23b4a9f3719e6db8eb1f753f71dd4f849c6101d13049f6091944914c31e6fe32db34daa176da230a04de12bf3';
+    const viewingPrivateKey = '0xeaa492b979aead8bdaf857673c4fe453a3f7ce5e8d661499391505f47d96d613';
+    const advancedMode = false;
+    const usePublicKeyChecked = false;
+    const encryptionCount = '661';
+
+    const ciphertext = encryptAccountData(
+      { recipientAddress, advancedMode, pubKey, usePublicKeyChecked },
+      { encryptionCount, viewingPrivateKey }
+    );
+    const decryptedData = decryptData(ciphertext, { encryptionCount, viewingPrivateKey });
+    expect(decryptedData).toEqual({
+      advancedMode,
+      usePublicKeyChecked,
+      address: recipientAddress,
+      pubKey: `0x99${pubKey.slice(4, 4 + 22)}`,
+    });
+  });
+});
+
+describe('storeSend', () => {
+  const testingUtils = generateTestingUtils({ providerType: 'default' });
+  const storeSendArgs = {
+    chainId: 5,
+    provider: new ethers.providers.Web3Provider(testingUtils.getProvider()),
+    viewingPrivateKey,
+    unencryptedAccountSendData: {
+      amount: '100',
+      tokenAddress,
+      txHash,
+      senderAddress: recipientAddress,
+    },
+    accountDataToEncrypt: {
+      recipientAddress: '0xEAC5F0d4A9a45E1f9FdD0e7e2882e9f60E301156',
+      advancedMode: false,
+      usePublicKeyChecked: false,
+      pubKey,
+    },
+  };
+  beforeEach(async () => {
+    testingUtils.clearAllMocks();
+    await localforage.clear();
+  });
+
+  it('Generates and saves an encryption count when there are no existing values and count', async () => {
+    const existingCount = await localforage.getItem(localStorageCountKey);
+    const value = await localforage.getItem(localStorageValueKey);
+    expect(existingCount).toEqual(undefined);
+    expect(value).toEqual(undefined);
+
+    await storeSend(storeSendArgs);
+
+    const newCount = await localforage.getItem(localStorageCountKey);
+    const values = (await localforage.getItem(localStorageValueKey)) as any[];
+    expect(newCount).toBeTruthy();
```

Comment:
Can we use stronger assertions than `toBeTruthy`? Like expect the actual number, or if we don't know it, expect that it is an integer >= 0

Reply:
https://github.com/ScopeLift/umbra-protocol/pull/494/commits/23e95594444466c2dd65e2834dfe95ed665f79db Ignore the previous change this one is better.